backend/indexer/internal/indexer/rpc.go

```diff
@@ -52,26 +42,15 @@ func (i *Indexer) getCurrentHeightWithRetries() (uint64, error) {
 func (i *Indexer) getBlockWithTransactions(height uint64, withTxs bool) (*types.EthBlock, error) {
 	var rpcResponse client.RPCResponse
 	blockNum := toBlockNumArg(height)
-	for retry := range make([]int, maxRetries) {
-		err := i.client.CallRpc("Filecoin.EthGetBlockByNumber", []interface{}{blockNum, withTxs}, &rpcResponse)
-		if err != nil {
-			return nil, fmt.Errorf("failed to execute rpc call: %w", err)
-		}
-		rpcErr := i.anyRPCError([]client.RPCResponse{rpcResponse})
-		if rpcErr == nil {
-			break
-		}
-		if retry == maxRetries-1 {
-			return nil, fmt.Errorf("rpc failure after %d retries: %s", maxRetries, rpcErr.Message)
-		}
-		backoffTime := time.Second * time.Duration(1<<uint(retry))
-		time.Sleep(backoffTime)
+	rpcResponses, err := i.retryRPCBatched([]string{"Filecoin.EthGetBlockByNumber"}, [][]interface{}{{blockNum, withTxs}})
+	if err != nil {
+		return nil, fmt.Errorf("%w", err)
 	}
+	rpcResponse = rpcResponses[0]
```

Comment:
Is it ever the case a rpcResonse with len > 1 is returned? Or an empty response?

Reply:
Not with err != nil and input arrays of length 1

Reply:
len check present in `CallRpcBatched`